proposals/dao/collateralizationOracle.ts

```diff
@@ -0,0 +1,486 @@
+      chainlinkEthUsdOracleWrapper,
+      chainlinkRaiUsdCompositOracle,
+      zeroConstantOracle.address,
+      oneConstantOracle.address
+    ]
+  );
+
+  logging && console.log('Collateralization Oracle: ', collateralizationOracle.address);
+
+  // 30. Collateralization Ratio Oracle Wrapper Implementation
+  const collateralizationOracleWrapperImplFactory = await ethers.getContractFactory('CollateralizationOracleWrapper');
+  const collateralizationOracleWrapperImpl = await collateralizationOracleWrapperImplFactory.deploy(
+    core,
+    1 // not used
+  );
+
+  logging && console.log('Collateralization Oracle Wrapper Impl: ', collateralizationOracleWrapperImpl.address);
+
+  // 31. Collateralization Ratio Oracle Wrapper Proxy
+  // This initialize calldata gets atomically executed against the impl logic
+  // upon construction of the proxy
+  const collateralizationOracleWrapperInterface = collateralizationOracleWrapperImpl.interface;
+  const calldata = collateralizationOracleWrapperInterface.encodeFunctionData('initialize', [
+    core,
+    collateralizationOracle.address,
+    CR_WRAPPER_DURATION,
+    CR_WRAPPER_DEVIATION_BPS
+  ]);
+
+  const ProxyFactory = await ethers.getContractFactory('TransparentUpgradeableProxy');
+  const proxy = await ProxyFactory.deploy(collateralizationOracleWrapperImpl.address, proxyAdmin, calldata);
+
+  const collateralizationOracleWrapper = await ethers.getContractAt('CollateralizationOracleWrapper', proxy.address);
+
+  logging && console.log('Collateralization Oracle Wrapper Proxy: ', collateralizationOracleWrapper.address);
+
+  return {
+    staticPcvDepositWrapper,
+    collateralizationOracle,
+    collateralizationOracleWrapper
+  } as NamedContracts;
+};
+
+export const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {};
+
+export const run: RunUpgradeFunc = async (addresses, oldContracts, contracts, logging = false) => {};
+
+export const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {};
+
+export const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts) => {
+  
+  // @ts-ignore
+  const staticPcvDepositWrapper: StaticPCVDepositWrapper = contracts.staticPcvDepositWrapper;
+  // @ts-ignore
+  const collateralizationOracle: CollateralizationOracle = contracts.collateralizationOracle;
+  // @ts-ignore
+  const collateralizationOracleWrapper: CollateralizationOracleWrapper = contracts.collateralizationOracleWrapper;
+
+  const guardian = addresses.multisig;
+
+  const { dai, weth, dpi, rai, fei} = addresses;
+
+  let staticBalances = await staticPcvDepositWrapper.resistantBalanceAndFei();
+  expect(staticBalances[0]).to.be.equal(ethers.constants.WeiPerEther.mul(4_000_000));
+  expect(staticBalances[1]).to.be.equal(ethers.constants.WeiPerEther.mul(11_500_000));
+
+  const tokens = await collateralizationOracle.getTokensInPcv();
+  expect(tokens.length).to.be.equal(6);
```

Comment:
I dislike the fact that this reports 6 tokens even though there's only actually five - the "USD" fake-token shouldn't count imo. @Joeysantoro is there any reason not to have this function report -1 on the "total"?

Reply:
It returns a list so no, we shouldn't exclude USD. We want to know how many deposits are abstractly USD